crates/wasi-http/src/types.rs

```diff
@@ -136,11 +138,34 @@ pub struct OutgoingRequestConfig {
 ///
 /// This implementation is used by the `wasi:http/outgoing-handler` interface
 /// default implementation.
+#[instrument(skip_all, fields(otel.kind = "client", url.full = Empty, http.request.method = Empty, http.response.status_code = Empty, otel.name = Empty, server.address = Empty, server.port = Empty))]
```

Comment:
I've not used `#[instrument]` myself before, but would it be possible to do something like `url.full = request.uri()` here in the macro instead of in the body below? I'm not sure if that has type restrictions on the return value of `.ur()`, but adding `.to_string()` may work well too.

Also, mind adding some comments as to what `otel.kind`, `otel.name`, and `server.{address,port}` are doing? The first two seem like sort of "standard variables" to set, but the second two aren't set here in the method either.

Reply:
Sure, I'll look into that @alexcrichton. Also sorry, I force pushed a linting fix without realizing you'd already started reviewing this.